lib/galaxy/jobs/__init__.py

```diff
@@ -366,7 +366,7 @@ def __init__(self, app: MinimalManagerApp):
             else:
                 job_config_file = self.app.config.job_config_file
                 if not self.app.config.is_set("job_config_file") and not os.path.exists(job_config_file):
-                    job_config_file = os.path.join(os.path.dirname(self.app.config.config_file), "job_conf.xml")
+                    job_config_file = os.path.join(self.app.config.config_dir, "job_conf.xml")
```

Comment:
I don't think this whole "if" clause is needed. I could be misreading it, but here's what I see:
"if `job_config_file` was not set (which means it's the default) AND its current value points to a nonexistent path (so the default path does not resolve to a file), then *set it to its default value* (but we've just checked it!).
And in the subsequent line (370) we check if that path exists - which we have done in line 368!
ping @natefoo ?

The value of `job_config_file` does not change in `config/__init__.py` if it is not set. If it *is* set, and we can't find it, we check against root - but that's it.

Reply:
@jdavcs I guess what you are missing is that the default is not `job_conf.xml` any more, but `job_conf.yml`, so we are checking 2 different things.

Reply:
Oh, I see! Now it makes sense. Thank you :)

Reply:
Right, I guess one should see the warning irrespective of the file's existence